### ISSUE TITLE
Work in conjunction with SPB to determine why calacs.e fails 

### ISSUE THREAD

**Issue by stscijgbot-hstdp** [spacetelescope/drizzlepac#1206]
_Issue [HLA-673](https://jira.stsci.edu/browse/HLA-673) was created on JIRA by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena):_

The calacs.e blows up when processing WFC (perhaps all CCD) data.  See ticket --HLA-666 for further information.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-673?focusedCommentId=565750#comment-565750):_

Processed WFC3 (IR and UVIS) and ACS (SBC, HRC, and WFC) data on both Linux (RHEL 7) and MacOS (10.15.7). These processings were not done for compatibility, but to determine which datasets fail in the condaforge processing.  The ACS CCD datasets (WFC and HRC) fail on the MacOS only at this time.

The processing fails after BLEVCORR and the *blv_tmp.fits file is never written.  As such, the CRCORR step cannot be done.  The issue is to determine where and why the *blv_tmp.fits file cannot be written.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-673?focusedCommentId=565999#comment-565999):_

I fixed a bug which was triggered ONLY when executing the condaforge distribution on MacOS for ACS CCD detectors where the failure was due to a missing return statement. While just adding a return statement (nothing really to return – not even status) would have resolved the issue, I chose to make a better solution which required some additional changes:
 (1) Only call computeDarktime() once to compute the updated DARKTIME primary keyword in contrast to being call for every imset where the computations were redundant, and 
 (2) Update
```java
int computeDarktime(ACSInfo *, SingleGroup *);```
to
```java
void computeDarktime(ACSInfo *, float *);```
Not only is the return now "void", but a reference to a SingleGroup object is not longer passed as this was unnecessary and potentially bad for the data.  A float variable is passed which is modified to contain the final darktime value.  This value is then used to update the DARKTIME FITS keyword in the PHDU of the output file associated with the SingleGroup object.

I then ran the ACS PyTests for HSTCAL, pushed the reviewed code to master, and tagged the code with "2.7.3rc1  Update for Condaforge Testing". 

I slacked Brendan (12 Jan 2022) to make a new Condaforge YML using Python 3.8 and the following based on the _*mvmalpha*_ release candidate.
```java
hstcal@2.7.3rc1
drizzlepac@3.4.0rc2
calcos@3.3.12-rc1
stwcs@1.7.2
tweakwcs@0.7.3
stsci.image@2.3.5
stsci.tools@4.0.1```

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-673?focusedCommentId=565999#comment-565999):_

I fixed a bug which was triggered ONLY when executing the condaforge distribution on MacOS for ACS CCD detectors where the failure was due to a missing return statement. While just adding a return statement (nothing really to return – not even status) would have resolved the issue, I chose to make a better solution which required some additional changes:
 (1) Only call computeDarktime() once to compute the updated DARKTIME primary keyword in contrast to being call for every imset where the computations were redundant, and 
 (2) Update
```java
int computeDarktime(ACSInfo *, SingleGroup *);```
to
```java
void computeDarktime(ACSInfo *, float *);```
Not only is the return now "void", but a reference to a SingleGroup object is not longer passed as this was unnecessary and potentially bad for the data.  A float variable is passed which is modified to contain the final darktime value.  This value is then used to update the DARKTIME FITS keyword in the PHDU of the output file associated with the SingleGroup object.

I then ran the ACS PyTests for HSTCAL, pushed the reviewed code to master, and tagged the code with "2.7.3rc1  Update for Condaforge Testing". 

I slacked Brendan (12 Jan 2022) to make a new Condaforge YML using Python 3.8 and the following based on the _*mvmalpha*_ release candidate.
```java
hstcal@2.7.3rc1
drizzlepac@3.4.0rc2
calcos@3.3.12-rc1
stwcs@1.7.2
tweakwcs@0.7.3
stsci.image@2.3.5
stsci.tools@4.0.1```
